packages/analytics-js/src/components/configManager/ConfigManager.ts

```diff
@@ -1,21 +1,148 @@
+/* eslint-disable class-methods-use-this */
+import { IHttpClient } from '@rudderstack/analytics-js/services/HttpClient/types';
+import { IErrorHandler } from '@rudderstack/analytics-js/services/ErrorHandler/types';
+import { ILogger } from '@rudderstack/analytics-js/services/Logger/types';
+import { defaultLogger } from '@rudderstack/analytics-js/services/Logger';
+import { defaultErrorHandler } from '@rudderstack/analytics-js/services/ErrorHandler';
+import { defaultHttpClient } from '@rudderstack/analytics-js/services/HttpClient';
+import { batch } from '@preact/signals-core';
+import { validateLoadArgs } from '@rudderstack/analytics-js/components/configManager/util/validate';
 import { state } from '@rudderstack/analytics-js/state';
+import { Destination } from '@rudderstack/analytics-js/state/types';
+import { resolveDataPlaneUrl } from './util/dataPlaneResolver';
+import { getIntegrationsCDNPath } from './util/cdnPaths';
+import { getSDKUrlInfo } from './util/commonUtil';
+import { IConfigManager, SourceConfigResponse } from './types';
+import { filterEnabledDestination } from './util/filterDestinations';
 
-class ConfigManager {
+class ConfigManager implements IConfigManager {
+  httpClient: IHttpClient;
+  errorHandler?: IErrorHandler;
+  logger?: ILogger;
+  hasErrorHandler = false;
+  hasLogger = false;
+
+  constructor(httpClient: IHttpClient, errorHandler?: IErrorHandler, logger?: ILogger) {
+    this.errorHandler = errorHandler;
+    this.logger = logger;
+    this.httpClient = httpClient;
+    this.hasErrorHandler = Boolean(this.errorHandler);
+    this.hasLogger = Boolean(this.logger);
+  }
+
+  /**
+   * A function to validate, construct and store loadOption, lifecycle, source and destination
+   * config related information in global state
+   */
   init() {
-    // TODO: get writekey, dataplaneUrl and loadOptions from state to use
+    validateLoadArgs(state.lifecycle.writeKey.value, state.lifecycle.dataPlaneUrl.value);
+    // determine the sourceConfig url
+    const intgCdnUrl = getIntegrationsCDNPath(
+      'process.package_version',
+      state.loadOptions.value.lockIntegrationsVersion as boolean,
+      state.loadOptions.value.destSDKBaseURL,
+    );
+
+    // determine if the staging SDK is being used
+    // TODO: deprecate this in new version and stop adding '-staging' in filenames
+    const { isStaging } = getSDKUrlInfo();
+
+    // set the final load option in state
+    batch(() => {
+      // set application lifecycle state in state
+      if (state.loadOptions.value.logLevel) {
+        state.lifecycle.logLevel.value = state.loadOptions.value.logLevel;
+      }
```

Comment:
Do we have to validate? It is already initialized with a default value, right?

Reply:
Even though it is initialized with default value we need to validate here. Else it complains.